### ISSUE TITLE
Regarding z-index

### ISSUE THREAD

**Issue by elgs** [xieziyu/angular2-draggable#18]
Thanks for the wonderful work. I really enjoyed reading your code.

I'm wondering whether it is possible to introduce a new input parameter for the `z-index`. Now when the drag stopped, the z-index is reset to its original value. To make it more intuitive in a desktop windows like environment, I'm hoping when the dragging is stopped, the most current dragged window would be on top of other windows. So in this case, users of the directive could keep a record of z-index somewhere else, and this count will be increased by 1 and be assigned to the most recent dragged window's `z-index` value.

**Comment by DanielCaspers** [comment]
For what its worth, I thought to do the same as you and bake that into the directive, but I took a different approach in the end. I created a component using the same draggable host listeners and modified the code to pull from a static count whenever mousedown on element, but global decrement from all on document mouseup.

You might find it more advantageous for layering most recently used in the way I've described. Otherwise each z index will need to be maintained in a map or graph structure outside of where they are used.
